From a3224a9da1da729753c2d5b54843fd1d4e5c88ab Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:00 -0400
Subject: [tools] perf: Fix tags/TAGS targets rebuilding

Message-id: <1408953210-26343-53-git-send-email-jolsa@redhat.com>
Patchwork-id: 88913
O-Subject: [PATCH RHEL7.1 BZ1133083 052/322] perf tools: Fix tags/TAGS targets rebuilding
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 26286141a43251bed548349efbf60b61f1c433e8
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Nov 26 13:54:12 2013 +0100

description
===========
Once the tags/TAGS file is generated it's never rebuilt until it's
removed by hand.

The reason is that the Makefile does not treat tags/TAGS as targets but
as files and thus won't rebuilt them once they are in place.

Adding PHONY tags/TAGS targets into Makefile.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/Makefile b/tools/perf/Makefile
index 4835618..eefb9fb 100644
--- a/tools/perf/Makefile
+++ b/tools/perf/Makefile
@@ -60,8 +60,11 @@ endef
 
 #
 # Needed if no target specified:
+# (Except for tags and TAGS targets. The reason is that the
+# Makefile does not treat tags/TAGS as targets but as files
+# and thus won't rebuilt them once they are in place.)
 #
-all:
+all tags TAGS:
  $(print_msg)
  $(make)
 
@@ -77,3 +80,5 @@ clean:
 %:
  $(print_msg)
  $(make)
+
+.PHONY: tags TAGS
-- 
1.7.1